// azure-search-types/dist/data-sources.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var DataSourceChangePolicy;
(function (DataSourceChangePolicy) {
    DataSourceChangePolicy["highWaterMark"] = "#Microsoft.Azure.Search.HighWaterMarkChangeDetectionPolicy";
    DataSourceChangePolicy["sqlIntegrated"] = "#Microsoft.Azure.Search.SqlIntegratedChangeTrackingPolicy";
})(DataSourceChangePolicy = exports.DataSourceChangePolicy || (exports.DataSourceChangePolicy = {}));
var DataSourceDeletePolicy;
(function (DataSourceDeletePolicy) {
    DataSourceDeletePolicy["soft"] = "#Microsoft.Azure.Search.SoftDeleteColumnDeletionDetectionPolicy";
})(DataSourceDeletePolicy = exports.DataSourceDeletePolicy || (exports.DataSourceDeletePolicy = {}));
var DataSourceType;
(function (DataSourceType) {
    DataSourceType["sql"] = "azuresql";
    DataSourceType["documentDb"] = "documentdb";
    DataSourceType["blob"] = "azureblob";
    DataSourceType["table"] = "azuretable";
})(DataSourceType = exports.DataSourceType || (exports.DataSourceType = {}));
