// azure-search-types/dist/skill-sets/vision/image-analysis.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ImageAnalysisInput;
(function (ImageAnalysisInput) {
    ImageAnalysisInput["image"] = "image";
})(ImageAnalysisInput = exports.ImageAnalysisInput || (exports.ImageAnalysisInput = {}));
var ImageAnalysisOutput;
(function (ImageAnalysisOutput) {
    ImageAnalysisOutput["categories"] = "categories";
    ImageAnalysisOutput["tags"] = "tags";
    ImageAnalysisOutput["description"] = "description";
    ImageAnalysisOutput["faces"] = "faces";
    ImageAnalysisOutput["imageType"] = "imageType";
    ImageAnalysisOutput["color"] = "color";
    ImageAnalysisOutput["adult"] = "adult";
})(ImageAnalysisOutput = exports.ImageAnalysisOutput || (exports.ImageAnalysisOutput = {}));
var VisualFeature;
(function (VisualFeature) {
    VisualFeature["categories"] = "categories";
    VisualFeature["tags"] = "tags";
    VisualFeature["description"] = "Description";
    VisualFeature["faces"] = "Faces";
    VisualFeature["imageType"] = "ImageType";
    VisualFeature["color"] = "Color";
    VisualFeature["adult"] = "Adult";
})(VisualFeature = exports.VisualFeature || (exports.VisualFeature = {}));
var ImageDetail;
(function (ImageDetail) {
    ImageDetail[ImageDetail["Celebrities"] = 0] = "Celebrities";
    ImageDetail[ImageDetail["Landmarks"] = 1] = "Landmarks";
})(ImageDetail = exports.ImageDetail || (exports.ImageDetail = {}));
