// azure-search-types/dist/skill-sets/text/entity-recognition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var EntityRecognitionInput;
(function (EntityRecognitionInput) {
    EntityRecognitionInput["languageCode"] = "languageCode";
    EntityRecognitionInput["text"] = "text";
})(EntityRecognitionInput = exports.EntityRecognitionInput || (exports.EntityRecognitionInput = {}));
var EntityRecognitionOutput;
(function (EntityRecognitionOutput) {
    EntityRecognitionOutput["persons"] = "persons";
    EntityRecognitionOutput["locations"] = "locations";
    EntityRecognitionOutput["organizations"] = "organizations";
    EntityRecognitionOutput["quantities"] = "quantities";
    EntityRecognitionOutput["dateTimes"] = "dateTimes";
    EntityRecognitionOutput["urls"] = "urls";
    EntityRecognitionOutput["emails"] = "emails";
    EntityRecognitionOutput["namedEntities"] = "namedEntities";
    EntityRecognitionOutput["entities"] = "entities";
})(EntityRecognitionOutput = exports.EntityRecognitionOutput || (exports.EntityRecognitionOutput = {}));
var Category;
(function (Category) {
    Category["person"] = "Person";
    Category["location"] = "Location";
    Category["organization"] = "Organization";
    Category["quantity"] = "Quantity";
    Category["datetime"] = "Datetime";
    Category["url"] = "URL";
    Category["email"] = "Email";
})(Category = exports.Category || (exports.Category = {}));
