// azure-search-types/dist/skill-sets/text/named-entity-recognition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var NamedEntityRecognitionInput;
(function (NamedEntityRecognitionInput) {
    NamedEntityRecognitionInput["languageCode"] = "languageCode";
    NamedEntityRecognitionInput["text"] = "text";
})(NamedEntityRecognitionInput = exports.NamedEntityRecognitionInput || (exports.NamedEntityRecognitionInput = {}));
var NamedEntityRecognitionOutput;
(function (NamedEntityRecognitionOutput) {
    NamedEntityRecognitionOutput["persons"] = "persons";
    NamedEntityRecognitionOutput["locations"] = "locations";
    NamedEntityRecognitionOutput["organizations"] = "organizations";
    NamedEntityRecognitionOutput["entities"] = "entities";
})(NamedEntityRecognitionOutput = exports.NamedEntityRecognitionOutput || (exports.NamedEntityRecognitionOutput = {}));
var NamedEntityCategory;
(function (NamedEntityCategory) {
    NamedEntityCategory["person"] = "Person";
    NamedEntityCategory["location"] = "Location";
    NamedEntityCategory["organization"] = "Organization";
})(NamedEntityCategory = exports.NamedEntityCategory || (exports.NamedEntityCategory = {}));
