// azure-search-types/dist/indexes/schema/analyzers/analyzers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var AnalyzerType;
(function (AnalyzerType) {
    AnalyzerType["pattern"] = "#Microsoft.Azure.Search.PatternAnalyzer";
    AnalyzerType["standard"] = "#Microsoft.Azure.Search.StandardAnalyzer";
    AnalyzerType["stop"] = "#Microsoft.Azure.Search.StopAnalyzer";
    AnalyzerType["custom"] = "#Microsoft.Azure.Search.CustomAnalyzer";
})(AnalyzerType = exports.AnalyzerType || (exports.AnalyzerType = {}));
var PrebuiltAnalyzerName;
(function (PrebuiltAnalyzerName) {
    PrebuiltAnalyzerName["keyword"] = "keyword";
    PrebuiltAnalyzerName["pattern"] = "pattern";
    PrebuiltAnalyzerName["simple"] = "simple";
    PrebuiltAnalyzerName["standard"] = "standard.lucene";
    PrebuiltAnalyzerName["standardAsciiFolding"] = "standardasciifolding.lucene";
    PrebuiltAnalyzerName["stop"] = "stop";
    PrebuiltAnalyzerName["whitespace"] = "whitespace";
})(PrebuiltAnalyzerName = exports.PrebuiltAnalyzerName || (exports.PrebuiltAnalyzerName = {}));
var MicrosoftLanguageAnalyzerName;
(function (MicrosoftLanguageAnalyzerName) {
    MicrosoftLanguageAnalyzerName["arabic"] = "ar.microsoft";
    MicrosoftLanguageAnalyzerName["bangla"] = "bn.microsoft";
    MicrosoftLanguageAnalyzerName["bulgarian"] = "bg.microsoft";
    MicrosoftLanguageAnalyzerName["catalan"] = "ca.microsoft";
    MicrosoftLanguageAnalyzerName["chineseSimplified"] = "zh-Hans.microsoft";
    MicrosoftLanguageAnalyzerName["chineseTraditional"] = "zh-Hant.microsoft";
    MicrosoftLanguageAnalyzerName["croatian"] = "hr.microsoft";
    MicrosoftLanguageAnalyzerName["czech"] = "cs.microsoft";
    MicrosoftLanguageAnalyzerName["danish"] = "da.microsoft";
    MicrosoftLanguageAnalyzerName["dutch"] = "nl.microsoft";
    MicrosoftLanguageAnalyzerName["english"] = "en.microsoft";
    MicrosoftLanguageAnalyzerName["estonian"] = "et.microsoft";
    MicrosoftLanguageAnalyzerName["finnish"] = "fi.microsoft";
    MicrosoftLanguageAnalyzerName["french"] = "fr.microsoft";
    MicrosoftLanguageAnalyzerName["german"] = "de.microsoft";
    MicrosoftLanguageAnalyzerName["greek"] = "el.microsoft";
    MicrosoftLanguageAnalyzerName["gujarati"] = "gu.microsoft";
    MicrosoftLanguageAnalyzerName["hebrew"] = "he.microsoft";
    MicrosoftLanguageAnalyzerName["hindi"] = "hi.microsoft";
    MicrosoftLanguageAnalyzerName["hungarian"] = "hu.microsoft";
    MicrosoftLanguageAnalyzerName["icelandic"] = "is.microsoft";
    MicrosoftLanguageAnalyzerName["indonesian"] = "id.microsoft";
    MicrosoftLanguageAnalyzerName["italian"] = "it.microsoft";
    MicrosoftLanguageAnalyzerName["japanese"] = "ja.microsoft";
    MicrosoftLanguageAnalyzerName["kannada"] = "ka.microsoft";
    MicrosoftLanguageAnalyzerName["korean"] = "ko.microsoft";
    MicrosoftLanguageAnalyzerName["latvian"] = "lv.microsoft";
    MicrosoftLanguageAnalyzerName["lithuanian"] = "lt.microsoft";
    MicrosoftLanguageAnalyzerName["malayalam"] = "ml.microsoft";
    MicrosoftLanguageAnalyzerName["malay"] = "ms.microsoft";
    MicrosoftLanguageAnalyzerName["marathi"] = "mr.microsoft";
    MicrosoftLanguageAnalyzerName["norwegian"] = "nb.microsoft";
    MicrosoftLanguageAnalyzerName["polish"] = "pl.microsoft";
    MicrosoftLanguageAnalyzerName["portugueseBrazil"] = "pt-Br.microsoft";
    MicrosoftLanguageAnalyzerName["portuguesePortugal"] = "pt-Pt.microsoft";
    MicrosoftLanguageAnalyzerName["punjabi"] = "pa.microsoft";
    MicrosoftLanguageAnalyzerName["romanian"] = "ro.microsoft";
    MicrosoftLanguageAnalyzerName["russian"] = "ru.microsoft";
    MicrosoftLanguageAnalyzerName["serbianCyrillic"] = "sr-cyrillic.microsoft";
    MicrosoftLanguageAnalyzerName["serbianLatin"] = "sr-latin.microsoft";
    MicrosoftLanguageAnalyzerName["slovak"] = "sk.microsoft";
    MicrosoftLanguageAnalyzerName["slovenian"] = "sl.microsoft";
    MicrosoftLanguageAnalyzerName["spanish"] = "es.microsoft";
    MicrosoftLanguageAnalyzerName["swedish"] = "sv.microsoft";
    MicrosoftLanguageAnalyzerName["tamil"] = "ta.microsoft";
    MicrosoftLanguageAnalyzerName["telugu"] = "te.microsoft";
    MicrosoftLanguageAnalyzerName["thai"] = "th.microsoft";
    MicrosoftLanguageAnalyzerName["turkish"] = "tr.microsoft";
    MicrosoftLanguageAnalyzerName["ukrainian"] = "uk.microsoft";
    MicrosoftLanguageAnalyzerName["urdu"] = "ur.microsoft";
    MicrosoftLanguageAnalyzerName["vietnamese"] = "vi.microsoft";
})(MicrosoftLanguageAnalyzerName = exports.MicrosoftLanguageAnalyzerName || (exports.MicrosoftLanguageAnalyzerName = {}));
var LuceneLanguageAnalyzerName;
(function (LuceneLanguageAnalyzerName) {
    LuceneLanguageAnalyzerName["arabic"] = "ar.lucene";
    LuceneLanguageAnalyzerName["armenian"] = "hy.lucene";
    LuceneLanguageAnalyzerName["basque"] = "eu.lucene";
    LuceneLanguageAnalyzerName["bulgarian"] = "bg.lucene";
    LuceneLanguageAnalyzerName["catalan"] = "ca.lucene";
    LuceneLanguageAnalyzerName["chineseSimplified"] = "zh-Hans.lucene";
    LuceneLanguageAnalyzerName["chineseTraditional"] = "zh-Hant.lucene";
    LuceneLanguageAnalyzerName["czech"] = "cs.lucene";
    LuceneLanguageAnalyzerName["danish"] = "da.lucene";
    LuceneLanguageAnalyzerName["dutch"] = "nl.lucene";
    LuceneLanguageAnalyzerName["english"] = "en.lucene";
    LuceneLanguageAnalyzerName["finnish"] = "fi.lucene";
    LuceneLanguageAnalyzerName["french"] = "fr.lucene";
    LuceneLanguageAnalyzerName["galician"] = "gl.lucene";
    LuceneLanguageAnalyzerName["german"] = "de.lucene";
    LuceneLanguageAnalyzerName["greek"] = "el.lucene";
    LuceneLanguageAnalyzerName["hindi"] = "hi.lucene";
    LuceneLanguageAnalyzerName["hungarian"] = "hu.lucene";
    LuceneLanguageAnalyzerName["indonesian"] = "id.lucene";
    LuceneLanguageAnalyzerName["irish"] = "ga.lucene";
    LuceneLanguageAnalyzerName["italian"] = "it.lucene";
    LuceneLanguageAnalyzerName["japanese"] = "ja.lucene";
    LuceneLanguageAnalyzerName["korean"] = "ko.lucene";
    LuceneLanguageAnalyzerName["latvian"] = "lv.lucene";
    LuceneLanguageAnalyzerName["norwegian"] = "no.lucene";
    LuceneLanguageAnalyzerName["persian"] = "fa.lucene";
    LuceneLanguageAnalyzerName["polish"] = "pl.lucene";
    LuceneLanguageAnalyzerName["portugueseBrazil"] = "pt-Br.lucene";
    LuceneLanguageAnalyzerName["portuguesePortugal"] = "pt-Pt.lucene";
    LuceneLanguageAnalyzerName["romanian"] = "ro.lucene";
    LuceneLanguageAnalyzerName["russian"] = "ru.lucene";
    LuceneLanguageAnalyzerName["spanish"] = "es.lucene";
    LuceneLanguageAnalyzerName["swedish"] = "sv.lucene";
    LuceneLanguageAnalyzerName["thai"] = "th.lucene";
    LuceneLanguageAnalyzerName["turkish"] = "tr.lucene";
})(LuceneLanguageAnalyzerName = exports.LuceneLanguageAnalyzerName || (exports.LuceneLanguageAnalyzerName = {}));
